nixos/modules/services/networking/pounce.nix

```
@@ -0,0 +1,224 @@
+        example = "~/.config/pounce/pouncecert.pem";
+        description = ''
+          Path for TLS Certificate for CertFP. Needed
+          for logging to Pounce password-less.
+        '';
+      };
+
+      sslKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/letsencrypt/cert/private.key";
+        description = ''
+          Path to server TLS certificate key.
+        '';
+      };
+
+      sslCert = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/letsencrypt/cert/cert.crt";
+        description = ''
+           Path to server TLS certificate.
+        '';
+      };
+
+      certFP = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/pounce/libera-user.pem";
+        description = ''
+           Path to certFP certificate for login to IRC server
+        '';
+      };
+
+      # If you want to use password, you need to generate it with pounce -x
+      hashedPass = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+        '';
+      };
+
+      ircHost = mkOption {
+        type = types.str;
+        example = "irc.libera.chat";
+        default = "";
+        description = ''
+          IRC channel
+        '';
+      };
+
+      ircHostPort = mkOption {
+        type = types.int;
+        default = 6697;
+        description = ''
+        '';
+      };
+
+      joinChannels = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          List of channel to join at startup
+        '';
+      };
+
+      ircUser = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC user
+        '';
+      };
+
+      ircNick = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC nickname
+        '';
+      };
+
+      ircPass = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC password
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = "Extra configuration";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.pounce = {
+      description = "Pounce";
+      after = [ "network-online.target" ];
+      wantedBy  = [ "multi-user.target" ];
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = "${pkgs.pounce}/bin/pounce ${configFile}";
+        Restart = "always";
+        RuntimeDirectory = "pounce";
+        RuntimeDirectoryMode = "0700";
+      };
+    };
+
+    users.users = optionalAttrs (cfg.user == defaultUser) {
+      ${defaultUser} =
+        { description = "Pounce server daemon owner";
+          group = cfg.group;
+          home = cfg.dataDir;
+          isNormalUser = true;
```

```suggestion
          isSystemUser = true;
```
daemon users are almost always system users. it might even be possible to run the bouncer using the `DynamicUser` systemd directives, that would remove the need for a dedicated user+group entirely.

---

O after that `DynamicUser` seems fitting here as described in https://github.com/NixOS/rfcs/blob/master/rfcs/0052-dynamic-ids.md.

---

Ok so i can simply remove all the user/group stuffs and replace them with 
```
serviceConfig.DynamicUser = true;
```
?

---

you'll also have to set `StateDirectory` (which *does* eliminate the `dataDir` option we had previously associated with `RuntimeDirectory` by brainfart) so systemd can set the necessary permission on the data set during service startup.

---

I've started polishing the module. I am not too sure about how I should use ```settings``` as pointed out by @aanderse